### PR TITLE
User auth without .htpasswd

### PR DESCRIPTION
**Pull request by MyIgel**
Based on #8
A user authentication that does not require any .htpasswd files

### CODE REVIEW

File: config-default.php [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<?php
+<?php defined('SESAM') or die('Direct access permitted');
```
Comment (607011) [inline config-default.php:1]: Will change text to `'Direct access not permitted'` ;-)

File: lib/functions.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    if (!$success) {
+        $status = 'error';
+    }
+
+    $data = array_merge(array(
+        'status' => $status
+    ), $data);
+
+    header('Content-Type: text/json');
+    $json = json_encode($data, JSON_UNESCAPED_SLASHES);
```
Comment (607011) [inline lib/functions.php:32]: `JSON_UNESCAPED_SLASHES` introduced in PHP 5.4. Will revert this line to

```
    $json = json_encode($data);
    echo str_replace("\/", "/", $json);
```

for backwards compatibility.

**Comment by MyIgel** [comment]
PHP 5.4 is no longer maintained & 5.5's EOL is 
10 Jul 2016, so jou could really set the minimum php-version to at least 5.6, older versions are really not recommended to use

@see http://php.net/supported-versions.php

**Comment by 607011** [comment]
[More than half of the PHP installations uses PHP 5.3 or older](http://w3techs.com/technologies/details/pl-php/5/all) ;-) If that changes ctSESAM-server will be upgraded to rely on PHP 5.4 or newer.

File: config-default.php (hunk 1 above)
Comment (MyIgel) [inline config-default.php:1]: Arg, C&P :D

**Pull request merged**